internal/provider/resource_tfe_workspace.go

```diff
@@ -513,6 +532,21 @@ func resourceTFEWorkspaceRead(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("Error reading configuration of workspace %s: %w", id, err)
 	}
 
+	tagBindings, err := config.Client.Workspaces.ListTagBindings(ctx, workspace.ID)
+	if err != nil {
+		if errors.Is(err, tfe.ErrResourceNotFound) {
```

Comment:
This would be a problem for any Terraform Enterprise version that does not have this endpoint. Maybe this particular 404 should not remove the workspace from state